Packaging doesn't preserve symlinks, use os.symlink

While preparing dev release packages I stumbled upon `letsencrypt_apache` test errors that don't show up in our standard testing suite. The difference is that I packaged everything (using `python setup.py sdist`/`python setup.py bdist_wheel`), installed in fresh venv (`pip install *.tar.gz`) and run "standalone" `nostests`:
```

+ nosetests letsencrypt acme letsencrypt_apache letsencrypt_nginx letshelp_letsencrypt
..................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................FF........E.F...........F........................................................................................................
======================================================================
ERROR: Make sure all vhosts are being properly found.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/tmp.aEsCqPpghg/venv/local/lib/python2.7/site-packages/letsencrypt_apache/tests/configurator_test.py", line 101, in test_get_virtual_hosts
    raise Exception("Missed: %s" % vhost)  # pragma: no cover
Exception: Missed: File: /tmp/tmp7P5gMbtemp/debian_apache_2_4/two_vhost_80/apache2/sites-available/letsencrypt.conf
Vhost path: /files/tmp/tmp7P5gMbtemp/debian_apache_2_4/two_vhost_80/apache2/sites-available/letsencrypt.conf/VirtualHost
Addresses: *:80
Name: letsencrypt.demo
Aliases:
TLS Enabled: No
Site Enabled: No

======================================================================
FAIL: test_find_best_vhost (letsencrypt_apache.tests.configurator_test.TwoVhost80Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/tmp.aEsCqPpghg/venv/local/lib/python2.7/site-packages/letsencrypt_apache/tests/configurator_test.py", line 141, in test_find_best_vhost
    self.vh_truth[3], self.config._find_best_vhost("letsencrypt.demo"))
AssertionError: <letsencrypt_apache.obj.VirtualHost object at 0x7f1169064b10> != <letsencrypt_apache.obj.VirtualHost object at 0x7f1169064790>

======================================================================
FAIL: test_find_best_vhost_default (letsencrypt_apache.tests.configurator_test.TwoVhost80Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/tmp.aEsCqPpghg/venv/local/lib/python2.7/site-packages/letsencrypt_apache/tests/configurator_test.py", line 165, in test_find_best_vhost_default
    self.config._find_best_vhost("example.demo"), self.vh_truth[2])
AssertionError: <letsencrypt_apache.obj.VirtualHost object at 0x7f1169033210> != <letsencrypt_apache.obj.VirtualHost object at 0x7f1169033250>

======================================================================
FAIL: Test if site is enabled.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/tmp.aEsCqPpghg/venv/local/lib/python2.7/site-packages/letsencrypt_apache/tests/configurator_test.py", line 180, in test_is_site_enabled
    self.assertTrue(self.config.is_site_enabled(self.vh_truth[0].filep))
AssertionError: False is not true

======================================================================
FAIL: test_redirect_with_conflict (letsencrypt_apache.tests.configurator_test.TwoVhost80Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/tmp.aEsCqPpghg/venv/local/lib/python2.7/site-packages/letsencrypt_apache/tests/configurator_test.py", line 535, in test_redirect_with_conflict
    errors.PluginError, self.config._enable_redirect, ssl_vh, "")
AssertionError: PluginError not raised
-------------------- >> begin captured logging << --------------------
letsencrypt_apache.configurator: DEBUG: Did not find http version of ssl virtual host attempting to create
letsencrypt_apache.configurator: INFO: Created redirect file: le-redirect.conf
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 883 tests in 31.251s

FAILED (errors=1, failures=4)
```

Any ideas, @jdkasten?

--- a/letsencrypt-apache/letsencrypt_apache/configurator.py
+++ b/letsencrypt-apache/letsencrypt_apache/configurator.py
@@ -1,5 +1,6 @@
 """Apache Configuration based off of Augeas Configurator."""
 # pylint: disable=too-many-lines
+import filecmp
 import itertools
 import logging
 import os
@@ -945,9 +946,11 @@ class ApacheConfigurator(augeas_configurator.AugeasConfigurator):
         """
         enabled_dir = os.path.join(self.parser.root, "sites-enabled")
         for entry in os.listdir(enabled_dir):
-            if os.path.realpath(os.path.join(enabled_dir, entry)) == avail_fp:
-                return True
-
+            try:
+                if filecmp.cmp(avail_fp, os.path.join(enabled_dir, entry)):
+                    return True
+            except OSError:
+                pass
         return False
 
     def enable_site(self, vhost):

--- a/tools/dev-release.sh
+++ b/tools/dev-release.sh
@@ -88,8 +88,7 @@ mkdir ../kgs
 kgs="../kgs/$version"
 pip freeze | tee $kgs
 pip install nose
-# TODO: letsencrypt_apache fails due to symlink, c.f. #838
-nosetests letsencrypt $SUBPKGS || true
+nosetests letsencrypt $SUBPKGS
 
 echo "New root: $root"
 echo "KGS is at $root/kgs"

